x/superfluid/keeper/migrate_test.go

```diff
@@ -153,34 +166,66 @@ func (suite *KeeperTestSuite) TestRouteLockedBalancerToConcentratedMigration() {
 
 			suite.ValidateMigrateResult(
 				ctx,
-				positionId, balancerPooId, poolIdLeaving, clPoolId, poolIdEntering, originalGammLockId, newGammLockId,
+				positionId, balancerPooId, poolIdLeaving, clPoolId, poolIdEntering,
 				tc.percentOfSharesToMigrate,
 				*balancerLock,
 				joinPoolAmt,
 				balancerPoolShareOut, coinsToMigrate,
 				amount0, amount1,
 			)
 
-			// Additional checks if the original gamm lock was superfluid staked.
-			if tc.superfluidDelegated {
-				// Check if migration deleted intermediary account connection.
+			// If the lock was superfluid delegated:
+			if tc.superfluidDelegated && !tc.superfluidUndelegating {
+				if tc.percentOfSharesToMigrate.Equal(sdk.OneDec()) {
+					// If we migrated all the shares:
+
+					// The intermediary account connection to the old gamm lock should be deleted.
+					addr := superfluidKeeper.GetLockIdIntermediaryAccountConnection(ctx, originalGammLockId)
+					suite.Require().Equal(addr.String(), "")
+
+					// The synthetic lockup should be deleted.
+					_, err = lockupKeeper.GetSyntheticLockup(ctx, originalGammLockId, keeper.StakingSyntheticDenom(balancerLock.Coins[0].Denom, valAddr.String()))
+					suite.Require().Error(err)
+
+					// The delegation from the intermediary account holder does not exist.
+					delegation, found := stakingKeeper.GetDelegation(ctx, balancerIntermediaryAcc.GetAccAddress(), valAddr)
+					suite.Require().False(found, "expected no delegation, found delegation w/ %d shares", delegation.Shares)
+				} else if tc.percentOfSharesToMigrate.LT(sdk.OneDec()) {
+					// If we migrated part of the shares:
+					// The intermediary account connection to the old gamm lock should still be present.
+					addr := superfluidKeeper.GetLockIdIntermediaryAccountConnection(ctx, originalGammLockId)
+					suite.Require().Equal(balancerIntermediaryAcc.GetAccAddress().String(), addr.String())
+
+					// Check if migration deleted synthetic lockup.
+					_, err = lockupKeeper.GetSyntheticLockup(ctx, originalGammLockId, keeper.StakingSyntheticDenom(balancerLock.Coins[0].Denom, valAddr.String()))
+					suite.Require().NoError(err)
+
+					// The delegation from the intermediary account holder should still exist.
+					_, found := stakingKeeper.GetDelegation(ctx, balancerIntermediaryAcc.GetAccAddress(), valAddr)
+					suite.Require().True(found, "expected delegation, found delegation no delegation")
+				}
```

Comment:
Can we do deeper checks here for partial unlocks?
Some of the checks that comes into my mind are 
- Check if superfluid delegated amount is existing amount - coinsToMigrate
- Check if what's remaining in the lock is existing amount - coinsToMigrate
- Check if the new superfluid staked amount is equal to coinsToMigrate

and so on

Reply:
Expanded these checks here, good call for this :) https://github.com/osmosis-labs/osmosis/pull/5162/commits/d71ad6938e87343a8d5ec65aefc88cbaa7044d4e

Reply:
If partial unlocking, is there a way we can check that the amount to undelegate is succesfully in the queue of undelegation?

Reply:
Can we also add checks for the new lock that has been created if this was a partial unlock? Apologies if I'm being picky here, but since we're touching core staking logic, I thought we might as well want to be as safe as possible

Reply:
If partial unlocking, there is no queue of undelegation. With superfluid, when we undelegate, the delegation amount is immediately removed from the validator and instantly unbonded

Reply:
Added a check to bonded, unbonding, and normal lock tests that check the newly created lock. Please let me know if I should expand! https://github.com/osmosis-labs/osmosis/pull/5162/commits/1aa1f055621fe24ecfc0a91e6c7bd218a7236ac8